### PR TITLE
[llvm-readobj][COFF] Dump hybrid object for ARM64X files.

### PR DESCRIPTION
**Pull request by cjacek**
Hybrid ARM64X files may theoretically change anything in the object (it may change both the PE headers and each sections), so I just print everything for both the primary and hybrid object. In practice, changing things like section table would be questionable (and would probably fail at runtime), so I considered skipping it for the hybrid objects, but decided against it for consistency.

My primary reason to implement it was to be able to write proper tests for lld-link, but since then it has proven to be neat tool when working with ARM64X files.

### CODE REVIEW

File: llvm/tools/llvm-readobj/llvm-readobj.cpp [hunk 1]
```diff
@@ -360,8 +360,8 @@ createDumper(const ObjectFile &Obj, ScopedPrinter &Writer) {
 }
 
 /// Dumps the specified object file.
-static void dumpObject(ObjectFile &Obj, ScopedPrinter &Writer,
-                       const Archive *A = nullptr) {
+void ObjDumper::dumpObject(ObjectFile &Obj, ScopedPrinter &Writer,
```
Comment (jh7370) [inline llvm/tools/llvm-readobj/llvm-readobj.cpp:363]: I don't like this: you're polluting the generic interface with parameters that are specific to one file format. It's okay to check `Obj.isCOFF()` in this function, and I'm also okay with it becoming an `ObjDumper` member (if moved to ObjDumper.cpp), but the hybrid object check really needs to be within the COFF-specific code.

One alternative approach might be to make the second `dumpObject` call, then call `printCOFFHybridObject` after that call. A variation of this might even be to introduce a new high-level function `dumpCOFFHybridObject`, much like the existing `dumpCOFFImportFile` etc that does this. (NB: I haven't given this a thorough thinking through, so this might not quite work, but hopefully it makes the principle clear).

Comment (cjacek) [reply llvm/tools/llvm-readobj/llvm-readobj.cpp]: Thanks for the feedback. I tried moving `dumpObject` to `ObjDumper.cpp`, but that would require making a number of `opts` variables non-static and became a bit messy. Instead, I kept is static and changed it to be more similar to `dumpMachOUniversalBinary`: I used a separate branch in `dumpInput` for COFF files, where I call `dumpObject` twice in case for hybrid objects. That made the change quite simple.

Also, following the suggestion from #102071, I changed tests to use `--match-full-lines` and `--strict-whitespace` to make sure that indention works as expected.

**Review by jh7370 — COMMENTED** [review]
Common code areas look good to me. I'll leave someone else to review the COFF-specific code.

**Review by mstorsjo — APPROVED** [review]
The COFF bits look good to me, thanks James for the review of the rest of it!

**Comment by cjacek** [comment]
Merged, thanks for reviews!
